### PR TITLE
Add hint and documentation for protocol version mismatches.

### PR DESCRIPTION
**Pull request by dwsteele**

### CODE REVIEW

File: doc/xml/release.xml [hunk 1]
```diff
@@ -94,6 +105,17 @@
...
                 <release-improvement-list>
+                    <release-item>
+                        <commit subject="Add hint and documentation for protocol version mismatches."/>
+
+                        <release-item-contributor-list>
+                            <release-item-ideator id="loop-evgeny"/>
+                            <release-item-reviewer id="cynthia.shang"/>
+                        </release-item-contributor-list>
+
+                        <p>Add note that <backrest/> versions must match when running remotely.</p>
```
Comment (cmwshang) [inline doc/xml/release.xml:116]: Why is this section in here twice (lines 74 - 83 as well as here)?

File: doc/xml/user-guide.xml [hunk 2]
```diff
@@ -2337,6 +2337,8 @@
 
             <p>A new host named <host>repository</host> is created to store the cluster backups.</p>
 
+            <admonition type="note">The <backrest/> version installed on the {[host-repo1]} host must exactly match the version installed on the <postgres/> host.</admonition>
```
Comment (cmwshang) [inline doc/xml/user-guide.xml:2340]: Replace `{[host-repo1]}`  with `<host>repository</host>` - In reading through this section of the document, repo1 is never mentioned except here whereas repository is, so repo1 vs repository would just confuse things.

**Review by cmwshang — CHANGES_REQUESTED** [review]
Review complete.

File: doc/xml/release.xml (hunk 1 above)
Comment (dwsteele) [reply doc/xml/release.xml:116]: This commit changes both the core code and the docs so it gets two release notes.

File: doc/xml/user-guide.xml (hunk 2 above)
Comment (dwsteele) [reply doc/xml/user-guide.xml]: I had also come to that conclusion and still managed to get it wrong. Fixed now.

**Review by cmwshang — APPROVED** [review]

**Pull request merged**